internal/services/compute/parse/community_gallery_image.go

```diff
@@ -0,0 +1,86 @@
+package parse
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type CommunityGalleryImageId struct {
+	GalleryName string
+	ImageName   string
+}
+
+func NewCommunityGalleryImageID(GalleryName, ImageName string) CommunityGalleryImageId {
+	return CommunityGalleryImageId{
+		GalleryName: GalleryName,
+		ImageName:   ImageName,
+	}
+}
+
+func (id CommunityGalleryImageId) String() string {
+	segments := []string{
+		fmt.Sprintf("Gallery Name %q", id.GalleryName),
+		fmt.Sprintf("Image Name %q", id.ImageName),
+	}
+	segmentsStr := strings.Join(segments, " / ")
+	return fmt.Sprintf("%s: (%s)", "Community Gallery", segmentsStr)
+}
+
+func (id CommunityGalleryImageId) ID() string {
+	fmtString := "/CommunityGalleries/%s/images/%s"
+	return fmt.Sprintf(fmtString, id.GalleryName, id.ImageName)
+}
+
+// CommunityGalleryImageID parses a CommunityGalleryImage Unique ID into an CommunityGalleryImageId struct
+func CommunityGalleryImageID(input string) (*CommunityGalleryImageId, error) {
+	segments := make([]resourceids.Segment, 0)
+
+	segments = append(segments, resourceids.Segment{
+		FixedValue: utils.String("CommunityGalleries"),
```

Comment:
in ARM these are supposed to be `camelCased`, so presumably this should be too?

Reply:
This is a weird edge case because these are actually `Provider` elements, but since the service team lopped off the resource ID and then renamed them as unique ID's in the middle of the actual Resource ID it looks odd and feels weird... I will go ahead and `camelCase` these elements as you suggested in the provider for consistency internally.

Reply:
Fixed